### ISSUE TITLE
remove support for files/package.install files

### ISSUE THREAD

**Issue by agarwal** [ocaml/opam#716]
After working on #675, I wonder why there should be support for `files/package.install` files within a package. The recommendation is to dynamically generate this file as part of a package's build commands, and opam will look for it when the build completes.

The supposed benefit of also allowing it to be included in the `files/` sub-directory of a package is that people might not want to bother figuring out how to generate the file dynamically. Okay, I get it up to this point. But then, once this file is created by hand, why not include it statically in your package's distribution rather than in the opam package?

In other words, the logic can be simplified. Opam will always look for a `package.install` file in the root of the project after successful completion of the `build` rules. However, opam doesn't care how the file gets there. Maybe it always was there or maybe the `build` commands produced it.
